crates/revm/src/batch.rs

```diff
@@ -37,6 +40,9 @@ pub struct BlockBatchRecord {
     ///
     /// Empty implies that there is going to be addresses to include in the filter in a future
     /// block. None means there isn't any kind of configuration.
+    #[cfg(not(feature = "std"))]
+    pruning_address_filter: Option<(u64, BTreeSet<Address>)>,
+    #[cfg(feature = "std")]
```

Comment:
~~I'd prefer if we could use the `hashbrown` crate if we are in no_std.
the hashbrown crate has a hashset type that is no_std~~
we should be able to reuse the Hashset type from revm::precompile instead which is no_std

Reply:
Bring in the nitpicks! Will do!

Reply:
If further changes are needed please let me know